packages/trpc/routers/bookmarks.ts

```diff
@@ -204,6 +204,13 @@ export const bookmarksAppRouter = router({
         if (alreadyExists) {
           return { ...alreadyExists, alreadyExists: true };
         }
+      } else if (input.type === "text") {
+        if (!input.text) {
+          throw new TRPCError({
+            message: "Creating empty bookmarks is not allowed",
+            code: "BAD_REQUEST",
+          });
+        }
```

Comment:
We don't typically do input validation on the code. This kind of validation should live in the zod schema. However, I don't really think this is something worth enforcing on the server side anyways.